### PR TITLE
Imag counter/init

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Just some boilerplate by now, waiting for #149, #171, #174 and maybe #191.

- [X] waiting on #149
- [ ] waiting on #171 
- [x] waiting on #174 
- [ ] waiting on #191

### CODE REVIEW

File: libimagcounter/src/counter.rs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        &self.name
+    }
+
+    pub fn value(&self) -> u64 {
+        self.value
+    }
+
+}
+
+impl From<Entry> for Counter {
```
Comment (matthiasbeyer) [inline libimagcounter/src/counter.rs]: I wonder whether this is actually possible. AFAIK, I can only get an `&Entry` from the store, but not `Entry`, am I right, @TheNeikos?

Comment (TheNeikos) [inline libimagcounter/src/counter.rs]: Yup

**Comment by matthiasbeyer** [comment]
Wait on #207 for more features via `ArgGroup`s from `clap`.

**Comment by matthiasbeyer** [comment]
I don't want to wait any longer for the two PRs to be merged, so can I please get some review on this? Linking support will be added later on for this binary.

**Comment by neithernut** [comment]
Some documentation would be nice.

**Comment by neithernut** [comment]
Eg. some usecase

**Comment by matthiasbeyer** [comment]
@neithernut What would you like to be documented? The binary? The library?

**Comment by neithernut** [comment]
First of all, an (informal) use-case would be nice. More important, however, would be "module documentation" for which you already laid out the structure in `/doc/src`

**Comment by matthiasbeyer** [comment]
ACK, will add this!

And then I will file some issues on how to improve this module.

**Comment by TheNeikos** [comment]
Looks good to me

**Pull request merged**